pkg/chunk/local/boltdb_index_client.go

```diff
@@ -23,6 +22,9 @@ var bucketName = []byte("index")
 const (
 	separator      = "\000"
 	dbReloadPeriod = 10 * time.Minute
+
+	DBOperationRead = iota
+	DBOperationWrite
```

Comment:
Could you explain why we may want this change into Cortex, instead of having Loki implementing its own `BoltIndexClient` with the feature it needs, please?

Reply:
Without this change, boltdb client creates a new boltdb file even for just write requests which end up getting flushed to archiver. This also helps cortex not create boltdb files unnecessarily.